api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -112,6 +112,14 @@ message ExtAuthz {
   //
   repeated string typed_metadata_context_namespaces = 16;
 
+  // Specifies a list of route metadata namespaces whose values, if present, will be passed to the
+  // ext_authz service. :ref:`filter_metadata <envoy_v3_api_field_config.core.v3.Metadata.filter_metadata>` is passed as an opaque ``protobuf::Struct``.
+  repeated string route_metadata_context_namespaces = 20;
```

Comment:
I'm actually a bit confused (as an API reviewer and someone with experience in ext auth) on how the series of transforms occurs here. I think this gets added to the attribute context in the ext_authz request? But the attribute request metadata refers to this as dynamic metadata? Is this the same as the stream-level dynamic metadata?

At the very least, we need better docs here.

CC @kyessenov

Reply:
As discussed in #30252, we should treat request/connection dynamic metadata differently from route metadata.

It is transmit through attribute context in ext_authz request, but in a different field, other than the original `metadata_context`.
https://github.com/envoyproxy/envoy/pull/30477/files#diff-d5e4f508f08a9d9ca1c77377d37a1a746c66959da5358aa5d17a49821fadae0a

For docs, do you mean we need to explicitly tell users these namespaces for route_metadata will be saved in a different field in the request?

Reply:
@htuch This is referring to "route" xDS metadata, not the dynamic metadata. It's indeed confusing to call all of it metadata, which is why we thought a separate field might be better.

Reply:
https://github.com/envoyproxy/envoy/issues/30252#issuecomment-1766918389 suggests a separate field in the check request as well, not just in config.

Reply:
Updated the description of ext_authz API to point to [`route_metadata_context` at `attribute_context.proto`](https://github.com/envoyproxy/envoy/pull/30477/files#diff-d5e4f508f08a9d9ca1c77377d37a1a746c66959da5358aa5d17a49821fadae0a)

The presubmit failure should be caused by the `:ref: 'envoy_v3_api_field_service.auth.v3.checkrequest'`. I have no clue what the correct path to messages in `attribute_context.proto` is. Please educate me about how this ref path is constructed.

Reply:
Perfect, thanks.